addons/repo-proxy/common/src/main/resources/default-repo-proxy.conf

```diff
@@ -0,0 +1,13 @@
+[repo-proxy]
+# enabled: By default, the repo-proxy add-on is disabled.
+#
+#enabled=false
+
+# proxy: define the proxy rule for the repos. A valid rule is like this: proxy.$packageType.$type.$name=$packageType:$type:$name.
+# proxy.maven.hosted.from=maven:remote:to
```

Comment:
will `proxy.maven:hosted:from=maven:remote:to` not work? (putting colons on both sides of the `=` I mean).

Also, can we do `proxy.maven.hosted.*=maven:remote:*`?

Reply:
Will try.

Reply:
> will `proxy.maven:hosted:from=maven:remote:to` not work? (putting colons on both sides of the `=` I mean).
Today I've tried with this, but seems it does not work. Our config system will treat key as "proxy.maven", and value as "hosted:from=maven:remote:to". This means it also think the colon as the splitter for key and value. Weird. :(

Reply:
Okay, I think I was trying to follow properties file syntax (loosely) when I wrote that config library. Bummer.

Reply:
Yeah, I think using name only probably makes sense. There are some ambiguities though, such as when the target Indy (not the static proxy) has a group and hosted repo with the same name (like pnc-builds)...which one do we proxy? I'm not sure, TBH...it might require us to have some configuration for what to prefer, maybe just an ordering like:
```
proxy.prefer = hosted,group
# proxy.prefer = group,hosted
```
But if so, we will need to pre-check the existence of the preferred one so we can fail over to trying to proxy the other.

What if we just used the following to setup the proxied names:

```
proxy.include=build*
proxy.include=shared-imports
proxy.include=maven:hosted:pnc-builds
proxy.include=maven:group:DA*
```

That way, we could use the simplest form (name) or the full form, and use simple wildcards with each. That would resolve the proxy.prefer mess too.

Reply:
I'm a little confused here.  In my mind all these proxy happens in the static-proxy instance itself, but not against the target. What we should do is set a remote repo to point to a repo in target. For example, if we want to do proxy in hosted:pnc-builds in target, what we should do is:
1. Set remote:pnc-builds (or something else) which remote url is the hosted:pnc-builds in target
2. Set proxy-rule here to proxy hosted:pnc-builds to remote:pnc-builds in static-proxy
3. When request against on http://static-proxy/xxx/hosted/pnc-builds/xxx, it will be forwarded to http://static-proxy/xxx/remote/pnc-builds/xxx, which will access the url in its remote url settings(target is here).
So if this is the aim, I think we don't need to care about the existence of the repos on target, in this proxy addon, because it's should be done by user when setting the remote:pnc-builds in static-proxy. 
Or I missed some context? Please correct me, @jdcasey (Or maybe a standard workflow from PNC side is good to show the meaning)

Reply:
Ok. Seems I missed some original requirements in NOS-2438. I'll take a look at it and then decide how to do the whole story.